Lib/pathlib/_local.py

```diff
@@ -57,6 +57,39 @@ def __repr__(self):
         return "<{}.parents>".format(type(self._path).__name__)
 
 
+class _Copier(CopierBase):
+    """Copier class that uses fast OS copy routine where possible, and ensures
+    symlinks' target_is_directory argument is properly set on Windows.
+    """
+    __slots__ = ()
+
+    if copyfile:
+        def copy_file(self, source, target, metadata_keys, dir_entry=None):
+            """Copy the given file to the given target."""
+            try:
+                source = os.fspath(source)
+            except TypeError:
+                if not isinstance(source, PathBase):
+                    raise
+                CopierBase.copy_file(self, source, target, metadata_keys, dir_entry)
+            else:
+                copyfile(source, os.fspath(target))
```

Comment:
Given the above exception handling for `fspath`, is `target` guaranteed to inherit from `PurePath` here?